src/cli/parse-flags.ts

```diff
@@ -35,203 +54,197 @@ export const parseFlags = (args: string[], sys?: CompilerSystem): ConfigFlags =>
+      value = false;
+    } else if (alias && cmdArg === `-${alias}`) {
+      value = true;
     }
 
-    for (let i = 0; i < args.length; i++) {
-      const cmdArg = args[i];
-
-      if (cmdArg.startsWith(`--${stringName}=`)) {
-        const values = cmdArg.split('=');
-        values.shift();
-        flags[flagKey] = values.join('=');
-        knownArgs.push(cmdArg);
-      } else if (cmdArg === `--${stringName}`) {
-        flags[flagKey] = args[i + 1];
-        knownArgs.push(cmdArg);
-        knownArgs.push(args[i + 1]);
-      } else if (cmdArg === `--${flagKey}`) {
-        flags[flagKey] = args[i + 1];
-        knownArgs.push(cmdArg);
-        knownArgs.push(args[i + 1]);
-      } else if (cmdArg.startsWith(`--${flagKey}=`)) {
-        const values = cmdArg.split('=');
-        values.shift();
-        flags[flagKey] = values.join('=');
-        knownArgs.push(cmdArg);
-      } else if (alias) {
-        if (cmdArg.startsWith(`-${alias}=`)) {
-          const values = cmdArg.split('=');
-          values.shift();
-          flags[flagKey] = values.join('=');
-          knownArgs.push(cmdArg);
-        } else if (cmdArg === `-${alias}`) {
-          flags[flagKey] = args[i + 1];
-          knownArgs.push(args[i + 1]);
-        }
-      }
+    if (value !== undefined && cmdArg !== undefined) {
+      flags[configCaseName] = value;
+      flags.knownArgs!.push(cmdArg);
     }
   });
+};
 
-  NUMBER_ARG_OPTS.forEach((numberName) => {
-    const alias = ARG_OPTS_ALIASES[numberName];
-    const flagKey = configCase(numberName);
+/**
+ * Parse a string CLI argument
+ *
+ * @param flags the config flags object, while we'll modify
+ * @param args our CLI arguments
+ * @param configCaseName the argument we want to look at right now
+ */
+const parseStringArg = (flags: ConfigFlags, args: string[], configCaseName: StringCLIArg) => {
+  if (typeof flags[configCaseName] !== 'string') {
+    flags[configCaseName] = null;
+  }
 
-    if (typeof flags[flagKey] !== 'number') {
-      flags[flagKey] = null;
-    }
+  const { value, matchingArg } = getValue(args, configCaseName);
+
+  if (value !== undefined && matchingArg !== undefined) {
+    flags[configCaseName] = value;
+    flags.knownArgs!.push(matchingArg);
+    flags.knownArgs!.push(value);
+  }
+};
+
+/**
+ * Parse a number CLI argument
+ *
+ * @param flags the config flags object, while we'll modify
+ * @param args our CLI arguments
+ * @param configCaseName the argument we want to look at right now
+ */
+const parseNumberArg = (flags: ConfigFlags, args: string[], configCaseName: NumberCLIArg) => {
+  if (typeof flags[configCaseName] !== 'number') {
+    flags[configCaseName] = null;
+  }
 
-    for (let i = 0; i < args.length; i++) {
-      const cmdArg = args[i];
-
-      if (cmdArg.startsWith(`--${numberName}=`)) {
-        const values = cmdArg.split('=');
-        values.shift();
-        flags[flagKey] = parseInt(values.join(''), 10);
```

Comment:
We may be introducing a subtle bug here by not having a `.join()` call for the new `parseNumberArg` impl, but that seems pretty far-fetched to me.  I'm in favor of taking that risk here, especially if our tests are passing.